### DISCUSSION TITLE
MIDI devices problems

### DISCUSSION THREAD

**Discussion by giamby3000** [GrandOrgue/grandorgue#2032] [Help]
Hi everybody. 

Since severals years I have issues with 2 MIDI input devices (used together - pedal and keyboard with usb connection) and GrandOrgue, in Windows and in Linux too.

I open GO 
MIDI Devices are seen by GO
in the Initial MIDI Config I set correctly the devices and the two MIDI are listened fine, but when I return to organ or one device doesn't respond, or nothing respond ( I don't see keyboard red square  around the keys or the pedals  I play) and no sound.

I try to
disconnect and reconnect usb midi 
update the app to the last version 
install GO in Windows or in Linux 

but nothing change. 

I tried an old free version of Haupwerke and my MIDI devices usb adapters work very well without issues. 

with GO is a nightmare.

Any ideas? 

Thank you

**Comment by oleg68** [comment]
> Any ideas?

- File -> Settings -> MIDI devices
    - Do you see your midi input devices?
    - Are they selected?
- Audio/Midi -> Log MIDI events.
    - Do you see any events when you press keys on the keyboard or on the pedal?

**Reply by giamby3000** [reply]
In Linux:
my input devices are checked and in Log MIDI events they are correctly detected. But when I return into organ console a keyboard doesn't work , sometimes also the other keyboard doesn't work. It's a random behaviour. 
My configuration is the following:
1) MIDI MATE interface -> pedal 
2) Roland interface -> Manual 1 AND Manual 2 (detected on the same channel 2)

Solution: after several attempts , closing and reopen GO, cache deletion now all work fine (also with two manuals connected to the same MIDI interface and the same physical keyboard)


Windows:
same configuration of Linux case (input MIDI devices seen and checked). The event detection after several attempts fails and nothing works.
I tried to 
Disconnect USB interface
Disable and reenable USB MIDI interfaces
Change USB ports
restart GO
Cache delete
Question: in the MIDI devices list I have many MIDI Interface saved during the various try. In the event detection tab I select the actual interface but nothing change: is it possible to erase all the list and restart with a blank list of devices?


GO works fine but the Initial MIDI configuration gives many issues.

Thank you
